packages/jsii-rosetta/lib/languages/default.ts

```diff
@@ -153,6 +153,22 @@ export abstract class DefaultVisitor<C> implements AstHandler<C> {
       context.report(unsup, `Use of ${ts.SyntaxKind[unsup.kind]} in an object literal is not supported.`);
     }
 
+    const anyMembersFunctions = node.properties.some((p) =>
+      ts.isPropertyAssignment(p)
+        ? isExpressionOfFunctionType(context.typeChecker, p.initializer)
+        : ts.isShorthandPropertyAssignment(p)
+        ? isExpressionOfFunctionType(context.typeChecker, p.name)
+        : false,
+    );
+
+    const inferredType = context.inferredTypeOfExpression(node);
+    if ((inferredType && isJsiiProtocolType(context.typeChecker, inferredType)) || anyMembersFunctions) {
```

Comment:
What is this checking for? I understand the `anyMembersFunctions` piece, but don't understand what `isJsiiProtocolType` is checking for.

Reply:
"Is this type a non-struct `interface` type that comes from jsii"

* non-struct `interface`: we use `interface`s for 2 purposes, and we constantly have to make distinctions between them. The terms I commonly use are "structs" and "protocols" (though we also use data types for the first, and "behavioral interfaces" for the second).
* From jsii: you can also define interfaces straight up in the example itself, and those are not subject to the rewrite. Therefore this code also checks if the interface comes from a jsii-ified library.